elide-core/src/main/java/com/yahoo/elide/parsers/state/CollectionTerminalState.java

```diff
@@ -71,8 +71,11 @@ public CollectionTerminalState(Class<?> entityClass, Optional<PersistentResource
         DocumentProcessor includedProcessor = new IncludedProcessor();
         includedProcessor.execute(jsonApiDocument, collection, queryParams);
 
-        DocumentProcessor sortProcessor = new SortProcessor();
-        sortProcessor.execute(jsonApiDocument, collection, queryParams);
+        // if we are using pagination, then we have also applied sorting rules directly into the HQL, etc
+        if (requestScope.getPagination().isEmpty()) {
```

Comment:
I think we should just get rid of the Sort DocumentProcessor and instead move that functionality into the InMemoryDataStore.  I don't think you have to bite off fixing the InMemoryDataStore in this PR - but you could remove the processor and this check.